src/component/modal/searchModal.js

```diff
@@ -0,0 +1,76 @@
+import request from "../../api.js";
+import SearchResultItem from "./searchResultItem.js";
+
+export default class SearchModal {
+
+    constructor({ rootElement, setPage }) {
+        const searchBgElement = document.createElement("div");
+        const searchModalElement = document.createElement("div");
+        const headerElement = document.createElement("div");
+        const searchIcon = document.createElement("img");
+        const searchInput = document.createElement("input");
+        const hr = document.createElement("hr");
+        this.searchResultElement = document.createElement("div");
+
+        searchIcon.src = "../../../public/searchicon.png";
+        searchBgElement.className = "searchModalBackground";
+        searchModalElement.className = "searchModal";
+        headerElement.className = "searchHeader";
+        searchInput.placeholder = "단어를 입력하고 엔터키를 눌러주세요.";
+
+        rootElement.appendChild(searchBgElement);
+        searchBgElement.appendChild(searchModalElement);
+        searchModalElement.appendChild(headerElement);
+        headerElement.appendChild(searchIcon);
+        headerElement.appendChild(searchInput);
+        searchModalElement.appendChild(hr);
+        searchModalElement.appendChild(this.searchResultElement);
+
+        this.setEvent(searchBgElement, searchInput, searchModalElement, setPage);
+    }
+
+    async getSearchResult(findText) {
+        const result = await request("/documents", {
+            method: `GET`,
+        });
+        const searchResult = [];
+        const queue = [...result];
+
+        while (queue.length) {
+            const nowNode = queue.shift();
+            if (nowNode.title && nowNode.title.includes(findText)) {
+                searchResult.push({ title: nowNode.title, id: nowNode.id });
+            }
+            nowNode.documents.map((documentItem) => {
+                queue.unshift(documentItem);
+            });
+        }
+        return searchResult;
+    }
+
+    setEvent(searchBgElement, searchInput, searchModalElement, setPage) {
+        searchBgElement.addEventListener("click", (event) => {
+            if (event.target.className === "searchModalBackground") {
+                searchBgElement.style.display = "none";
+            }
+        })
+
+        searchInput.addEventListener("keydown", async (event) => {
+            if (event.keyCode !== 13)
+                return;
```

Comment:
13! 13이 엔터 키였군요. 한 번에 알기 어려우니 다음과 같이 쓰는 건 어떤가요? 또, 중괄호를 사용하지 않는다면 한 줄로 작성해주시는게 좋을 것 같습니다 : )
```suggestion
            if (event.key !== "Enter") return;
```

Reply:
저번에도 조건문 관련 같은 피드백 주셔서, 꾸준히 적용하려고하는데 아직 적응이 잘안되서 그런지 몇몇개는 자꾸 중괄호 없이나 개행하고 쓰게 되네요.